### PR TITLE
Get cache table details using `dry.` run of cached calls.

### PR DESCRIPTION
**Pull request by khalidharun**
Allow end users to pass a `dry.` parameter to inspect the cache table status without actually executing the calls.  Replaces #103

### CODE REVIEW

File: R/cache.R [hunk 1]
```diff
@@ -236,15 +236,20 @@ cache <- function(uncached_function, key, salt, con, prefix = deparse(uncached_f
...
 
-  ## Check "force." name collision
-  if ("force." %in% names(formals(cached_function))) {
-    stop(sQuote("force."), " is a reserved argument in caching layer, ",
-         "collision with formals in the cached function.", call. = FALSE)
-  }
+  ## Check "force.", "dry." name collision
+  lapply(c("force.", "dry."), function(x) {
+    if (x %in% names(formals(cached_function))) {
+      stop(sQuote(x), " is a reserved argument in caching layer, ",
```
Comment (kirillseva) [inline R/cache.R]: can you rephrase that it's cachemeifyoucan, not a general caching layer? you never know how people can use open source software

Comment (robertzk) [inline R/cache.R]: Good point.

File: R/cache.R [hunk 2]
```diff
@@ -317,15 +325,21 @@ build_cached_function <- function(cached_function) {
...
-      cachemeifyoucan:::cached_function_call(`_uncached_function`, call,
+    fcn_call <- cachemeifyoucan:::cached_function_call(`_uncached_function`, call,
         parent.frame(), tbl_name, `_key`, `_con`, force., `_batch_size`)
-    )
+
+    ## Grab the all keys
+    keys <- fcn_call$call[[fcn_call$key]]
+
+    ## Log cache metadata if in debug mode
+    status <- debug_info(fcn_call, keys)
```
Comment (robertzk) [inline R/cache.R:340]: `options(cachemeifyoucan.debug = TRUE)` should continue to print the debug info even if `dry. = FALSE`.

File: R/cache.R [hunk 3]
```diff
@@ -399,24 +412,42 @@ execute <- function(fcn_call) {
...
-        num_cols <- DBI::dbGetQuery(fcn_call$con,
-          paste0("select count(column_name) from information_schema.columns where table_name='", name, "'"))[1, 1]
-        message('  ', name, ': ', num_rows, ' rows * ', num_cols, ' columns')
-      } else {
-        message('  ', name, ': new shard')
-      }
-    })
-    message(length(cached_keys), ' cached keys')
-    message(length(uncached_keys), ' uncached keys')
+    msg <- pp(paste(
```
Comment (robertzk) [inline R/cache.R]: Please do not use `pp`, use `paste0` or `sprintf`.

Comment (rylandely) [inline R/cache.R]: Ever?

File: R/cache.R [hunk 4]
```diff
@@ -399,24 +412,42 @@ execute <- function(fcn_call) {
...
+        shard_info,
+        "#{ length(cached_keys) } cached keys",
+        "#{ length(uncached_keys) } uncached keys"
+      ),
+      collapse = "\n"
+    ))
+    message(msg)
   }
+
+  list(
```
Comment (robertzk) [inline R/cache.R:445]: Maybe add a class `cache_debug_info` so we can add a `print` method later?

File: R/cache.R (hunk 3 above)
Comment (kirillseva) [inline R/cache.R]: in open source packages

File: R/cache.R (hunk 1 above)
Comment (khalidharun) [inline R/cache.R]: :+1:

File: R/cache.R (hunk 2 above)
Comment (khalidharun) [inline R/cache.R:340]: It does continue to print because `debug_info(...)` is always called regardless of the value of the `is_dry`.

File: R/cache.R (hunk 3 above)
Comment (khalidharun) [inline R/cache.R]: :+1:

**Comment by khalidharun** [comment]
@robertzk Ready to merge.
